### PR TITLE
Task 15: Implement GraphQL mutations for replying to review comments

### PR DESCRIPTION
**Pull request by tonyblank**
## Summary
- Implement GraphQL mutation support for posting replies to pull request review comments
- Add hybrid approach using GraphQL for thread node IDs and REST for backward compatibility  
- Maintain complete CLI interface compatibility with zero breaking changes

## Key Changes
- **New GraphQL mutations**: `addPullRequestReviewThreadReply` for thread IDs (PRT_, PRRT_, RT_)
- **Hybrid implementation**: GraphQL for thread node IDs, REST API fallback for numeric/comment IDs
- **Comprehensive error handling**: GraphQL error mapping to appropriate exceptions
- **Full test coverage**: 28 new test cases covering all mutation scenarios and edge cases
- **Zero CLI changes**: Users experience no difference in interface or behavior

## Technical Implementation
- Created `reply_mutations.py` with GraphQL mutation builders
- Updated `ReplyService` with hybrid GraphQL/REST approach
- Added comprehensive test suite in `test_reply_mutations.py`
- Maintained all existing functionality and test compatibility

## Test Plan
- [x] All 664 existing tests pass
- [x] 28 new GraphQL mutation tests pass
- [x] `make check` passes (linting, type checking, formatting)
- [x] CLI interface unchanged - backward compatibility maintained
- [x] Error handling works for both GraphQL and REST scenarios

This implementation modernizes the reply functionality using GitHub's preferred GraphQL API while ensuring zero disruption to existing users.

### CODE REVIEW

File: src/toady/reply_service.py [hunk 1]
```diff
@@ -79,13 +88,212 @@ def post_reply(
...
+
+            if not comment_data:
+                raise ReplyServiceError(
+                    "No comment data returned from GraphQL mutation"
+                )
+
+            return self._build_reply_info_from_graphql(
+                comment_data, request, fetch_context, owner, repo
+            )
+
```
[the remark this answers is not in the document]
Comment (tonyblank) [reply src/toady/reply_service.py:199]: Great catch\! I've fixed this dead code issue by properly integrating the `_post_comment_reply` method into the `post_reply` flow.

**What was wrong:**
The `_post_comment_reply` method was implemented but never called because the routing logic only handled "thread_reply" strategy and fell back to REST API for everything else, including comment node IDs that should use GraphQL.

**How I fixed it:**
Updated the routing logic in `post_reply` (lines 95-105) to properly route comment node IDs to the GraphQL method:
- Thread node IDs (PRT_, PRRT_, RT_) → `_post_thread_reply` (GraphQL)
- Comment node IDs (IC_, PRRC_, RP_) → `_post_comment_reply` (GraphQL) 
- Numeric IDs → `_post_reply_fallback_rest` (REST for backward compatibility)

This ensures we use the modern GraphQL mutations for all node IDs while maintaining REST fallback for legacy numeric IDs. All tests are passing with this fix\!

**Learning:** Dead code often indicates incomplete integration rather than unnecessary code. The GraphQL comment reply functionality was valuable, it just needed proper wiring into the main flow.
